source/robot_model/test/tests/test_kinematics.cpp

```diff
@@ -0,0 +1,118 @@
+#include "robot_model/Model.hpp"
+
+#include <stdexcept>
+#include <memory>
+#include <gtest/gtest.h>
+
+#include "robot_model/exceptions/InvalidJointStateSizeException.hpp"
+#include "robot_model/exceptions/FrameNotFoundException.hpp"
+
+using namespace robot_model;
+
+class RobotModelKinematicsTest : public testing::Test {
+protected:
+  void SetUp() override {
+    robot_name = "franka";
+    urdf_path = std::string(TEST_FIXTURES) + "panda_arm.urdf";
+    franka = std::make_unique<Model>(robot_name, urdf_path);
+    joint_state = state_representation::JointState(robot_name, 7);
+  }
+
+  std::unique_ptr<Model> franka;
+  std::string robot_name;
+  std::string urdf_path;
+
+  state_representation::JointState joint_state;
+
+  double tol = 1e-5;
+
+// Known test configurations generated by fixtures/generateRobotModelKinematicsTestConfigurations.m
+  std::vector<state_representation::JointState> test_configs;
+  std::vector<std::vector<double>> test_fk_ee_expects;
+  std::vector<std::vector<double>> test_fk_link4_expects;
+  std::vector<std::vector<double>> test_velocity_fk_expects;
+
+  void setTestConfigurations() {
+    // Random test configuration 1:
+    state_representation::JointState config1("robot", 7);
+    config1.set_positions({-0.600782, -0.754768, 0.863711, -2.290588, 2.816429, 2.519369, 0.969137});
+    config1.set_velocities({0.578712, -0.861191, -0.513509, 0.019743, 0.731455, 0.994781, 0.219667});
+    test_configs.push_back(config1);
+
+    // Expected results for configuration 1:
+    test_fk_ee_expects.push_back({2.183865, 3.748338, -3.253473, 0.737074, -0.596918, -0.216162, -0.231700});
```

Comment:
Are those poses? Why not using a `std::vector<CartesianPose>` as you did with `JointState`? This way you can use `isApprox` function from `Eigen` on the `linear` and `angular` when testing expecting results. This saves some lines and make it a bit clrearer.

Reply:
I guess the relevant generator expression to keep it in one line would be:

```matlab
    fprintf('state_representation::CartesianPose fk_ee%i("ee", %s, %s);\n', ...
        conf, num2stdvec(ee_pose(1:3)), num2stdvec(ee_pose(4:7)));
```